addons/web_unsplash/static/src/js/unsplash_image_widget.js

```diff
@@ -159,7 +159,7 @@ widgetsMedia.ImageWidget.include({
      * @private
      */
     _loadMoreImages: function (forceSearch) {
-        if (!this.$('.o_we_search').val()) {
+        if (!this.$('.o_we_search').val() || this.searchService === "database") {
```

Comment:
Seems logical indeed to not do something different for `_loadMoreImages` when the search service is database only... but this feels like a half-fix: why is it not working when going through the `search` below? If we search for all images, why can we load more database images once all unsplash images have been loaded (can we?). Also why would the `search` override in unsplash not have the same condition? (if searchService is database then no call to unsplash)... I think this needs more checking.

Would also be great to add a test about this if possible :+1:

Reply:
Before anything else: this PR is only for 15.0 & 15.2. That whole widget was refactored with Owl during the front to backend refactor, and the bug of the OPW is fixed as of 16.0.

> why is it not working when going through the search below?

`search()` is designed only for the initial request. It fetches the attachments with a fixed limit (i.e. the initial 30, with a hard-coded offset of 0; then `fetchAttachments()` adds 1 more just to know if there is more). So the simple answer is: `search()` is just not designed for that.

> If we search for all images, why can we load more database images once all unsplash images have been loaded (can we?).

No, it doesn't work (until 16.0). But this seems to be intentional (cf. last bit in this [commit message](https://github.com/odoo/odoo/commit/c46204dff99c42a5b6f8b49da2094f850b619d6f)). There's this in the public `fetchAttachments()` of `ImageWidget` in `addons/web_editor/static/src/js/wysiwyg/widgets/media.js`:
```js
        if (this.needle && this.searchService !== 'database') {
            number = this.MAX_DB_ATTACHMENTS;
            offset = 0;
        }
```
What this effectively does is limit DB images to the 5 first results.

> Also why would the search override in unsplash not have the same condition?

Do you mean adding this before the code calling the unsplash API?
```js
if (!["all", "unsplash"].includes(this.searchService)) {
    return;
}
```

Reply:
> Do you mean adding this before the code calling the unsplash API?

Yes, but in the end, not required in stable given your explanations. Since this is only for 15.0 and 15.2, let's not overthink. I'll test and merge your PR as it is :+1: